src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/HtmlCSharp/CompletionResolveHandlerTest.cs

```diff
@@ -44,163 +53,142 @@ public CompletionResolveHandlerTest()
+                TriggerCharacters = CompletionHandler.AllTriggerCharacters.ToArray(),
+            }
+        };
 
         [Fact]
-        public async Task HandleRequestAsync_NullInsertTextAndTextEdit_RemapsResolvedTextEdit()
+        public async Task HandleRequestAsync_CSharpProjection_RemapsComplexTextEdit()
         {
             // Arrange
-            var originalEdit = new TextEdit() { NewText = "original" };
-            var mappedEdit = new TextEdit() { NewText = "mapped" };
-            DocumentMappingProvider.AddMapping(originalEdit, mappedEdit);
-            var requestedCompletionItem = new CompletionItem();
-            AssociateRequest(LanguageServerKind.CSharp, requestedCompletionItem, CompletionRequestContextCache, LoggerProvider);
-            var resolvedCompletionItem = new CompletionItem()
+            var text =
+                """
+                @using System
+                @code
+                {
+                    class C
+                    {
+                        override 
+                    }
+                }
+                """;
+
+            var cursorPosition = new Position(5, 17);
+            var documentUri = new Uri("C:/path/to/file.razor");
+            var completionParams = new CompletionParams()
             {
-                TextEdit = originalEdit,
+                TextDocument = new TextDocumentIdentifier() { Uri = documentUri },
+                Context = new VSInternalCompletionContext()
+                {
+                    TriggerKind = CompletionTriggerKind.TriggerCharacter,
+                    TriggerCharacter = " ",
+                    InvokeKind = VSInternalCompletionInvokeKind.Typing
+                },
+                Position = cursorPosition
             };
-            var requestInvoker = CreateRequestInvoker((method, languageServerName, completionItem) => resolvedCompletionItem);
-            var handler = new CompletionResolveHandler(requestInvoker, DocumentManager, DocumentMappingProvider, FormattingOptionsProvider, CompletionRequestContextCache, LoggerProvider);
 
             // Act
-            var result = await handler.HandleRequestAsync(requestedCompletionItem, new ClientCapabilities(), CancellationToken.None).ConfigureAwait(false);
+            var (unresolvedItem, resolvedItem, textEditRemapCount) = await ExecuteCSharpCompletionResolveRequestAsync(
+                documentUri, text, completionParams, itemLabel: "Equals(object obj)").ConfigureAwait(false);
 
             // Assert
-            Assert.Same(mappedEdit, result.TextEdit);
+            Assert.True(unresolvedItem.VsResolveTextEditOnCommit);
+            Assert.Null(unresolvedItem.TextEdit);
+            Assert.NotNull(resolvedItem.TextEdit);
+            Assert.Equal(1, textEditRemapCount);
         }
 
         [Fact]
-        public async Task HandleRequestAsync_NullInsertTextAndTextEdit_RemapsAdditionalTextEdits()
```

Comment:
The C# LSP server doesn't return additional text edits in any scenario today so this scenario was impossible (and imo unnecessary) to test